From 83feef2df5926709be95e518c9b7c998bf398bfd Mon Sep 17 00:00:00 2001
From: Richard Yu <richard.yu@mitrastar.com.tw>
Date: Sun, 4 May 2025 01:52:23 +0800
Subject: [PATCH 5/7] agent: handle_ap_autoconfiguration_response: Restrict
 agent with local controller to onboard only to its local controller

Ensure that when an agent has a designated local controller, it only accepts
AP Configuration Response messages from that specific controller. This prevents
the agent from mistakenly onboarding to another controller in the network.

https://prplfoundationcloud.atlassian.net/browse/PPM-3297

Signed-off-by: Richard Yu <richard.yu@mitrastar.com.tw>
---
 agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp b/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
index 5f3f07185..03c87a4b3 100644
--- a/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
+++ b/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
@@ -988,6 +988,11 @@ void ApAutoConfigurationTask::handle_ap_autoconfiguration_response(
     ieee1905_1::CmduMessageRx &cmdu_rx, const sMacAddr &src_mac)
 {
     auto db = AgentDB::get();
+    if (db->device_conf.local_controller && src_mac != db->bridge.mac) {
+        LOG(INFO) << "This agent has a local controller with mac=" << db->bridge.mac
+                  << " but response came from src_mac=" << src_mac << ", ignoring";
+        return;
+    }
     if (db->controller_info.bridge_mac != network_utils::ZERO_MAC &&
         src_mac != db->controller_info.bridge_mac) {
         LOG(INFO) << "current controller_bridge_mac=" << db->controller_info.bridge_mac
-- 
2.34.1

